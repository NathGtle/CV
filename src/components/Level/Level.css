.skillList {
  grid-row: 1;
  grid-column-end: span 5;
}

.skill {
  background-color:#FFFFFF;
  border-radius: 50%;
  display:inline-grid;
  height:13px;
  margin-left: 1px;
  width:13px;
}

.full {
  background-color:#6bbae4;
}

.thirty {
  background: linear-gradient(90deg, #6bbae4 25%, #FFFFFF 75%);
}

.half {
  background: linear-gradient(90deg, #6bbae4 50%, #FFFFFF 50%);
}

.seventy {
  background: linear-gradient(90deg, #6bbae4 75%, #FFFFFF 25%);
}
