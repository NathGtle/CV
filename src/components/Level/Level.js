import React from 'react'
import './Level.css'

const Level = ({ level }) => {
  const levelValue = level
  const decimal = Math.floor((levelValue - Math.floor(levelValue)) * 10)
  const max = 5
  const full = Math.floor(levelValue)
  const white = max - full - 1
  let decimalClass = 'skill half'
  switch (decimal) {
    case 0:
      decimalClass = 'skill'
      break
    case 1:
    case 2:
    case 3:
      decimalClass = 'skill thirty'
      break
    case 7:
    case 8:
    case 9:
      decimalClass = 'skill seventy'
      break
    default:
      decimalClass = 'skill half'
      break
  }
  return (
    <div className="skillList">
      {Array(full)
        .fill()
        .map((e, i) => (
          <span key={i} className="skill full" />
        ))}
      <span className={decimalClass} />
      {Array(white)
        .fill()
        .map((e, i) => (
          <span key={i} className="skill" />
        ))}
    </div>
  )
}

export default Level
