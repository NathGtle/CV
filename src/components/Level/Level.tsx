import React from 'react'
import PropTypes from 'prop-types'
import './Level.css'

interface LevelInterface {
  level: string
}

// TODO Refactor this sh**

const Level = ({ level }: LevelInterface) => {
  const levelValue = parseFloat(level)
  const decimal = Math.floor((levelValue - Math.floor(levelValue)) * 10)
  const max = 5
  const full = Math.floor(levelValue)
  const white = max - full - 1
  let decimalClass = 'skill half'
  switch (decimal) {
    case 0:
      decimalClass = 'skill'
      break
    case 1:
    case 2:
    case 3:
      decimalClass = 'skill thirty'
      break
    case 7:
    case 8:
    case 9:
      decimalClass = 'skill seventy'
      break
    default:
      decimalClass = 'skill half'
      break
  }
  return (
    <div className="skillList">
      {Array(full)
        .fill(0)
        .map((_e: number, i: number) => (
          <span key={i} className="skill full" />
        ))}
      <span className={decimalClass} />
      {Array(white)
        .fill(0)
        .map((_e: number, i: number) => (
          <span key={i} className="skill" />
        ))}
    </div>
  )
}

Level.propTypes = {
  level: PropTypes.string.isRequired,
}

export default Level
