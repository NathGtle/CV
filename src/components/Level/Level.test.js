import React from 'react'
import renderer from 'react-test-renderer'
import Level from './Level'

describe('Level component', () => {
  it('should render correctly full level complete', () => {
    const component = renderer.create(<Level level="4.9" />)

    // Trick avoid to make same TU for coverage branches
    component.update(<Level level="4.8" />)
    component.update(<Level level="4.7" />)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly half level', () => {
    const component = renderer.create(<Level level="2.5" />)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly thirty level', () => {
    const component = renderer.create(<Level level="2.2" />)

    // Trick avoid to make same TU for coverage branches
    component.update(<Level level="2.3" />)
    component.update(<Level level="2.1" />)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly empty level', () => {
    const component = renderer.create(<Level level="0" />)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
