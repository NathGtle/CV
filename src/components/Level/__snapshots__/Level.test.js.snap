// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level component should render correctly empty level 1`] = `
<div
  className="skillList"
>
  <span
    className="skill"
  />
  <span
    className="skill"
  />
  <span
    className="skill"
  />
  <span
    className="skill"
  />
  <span
    className="skill"
  />
</div>
`;

exports[`Level component should render correctly full level complete 1`] = `
<div
  className="skillList"
>
  <span
    className="skill full"
  />
  <span
    className="skill full"
  />
  <span
    className="skill full"
  />
  <span
    className="skill full"
  />
  <span
    className="skill seventy"
  />
</div>
`;

exports[`Level component should render correctly half level 1`] = `
<div
  className="skillList"
>
  <span
    className="skill full"
  />
  <span
    className="skill full"
  />
  <span
    className="skill half"
  />
  <span
    className="skill"
  />
  <span
    className="skill"
  />
</div>
`;

exports[`Level component should render correctly thirty level 1`] = `
<div
  className="skillList"
>
  <span
    className="skill full"
  />
  <span
    className="skill full"
  />
  <span
    className="skill thirty"
  />
  <span
    className="skill"
  />
  <span
    className="skill"
  />
</div>
`;
